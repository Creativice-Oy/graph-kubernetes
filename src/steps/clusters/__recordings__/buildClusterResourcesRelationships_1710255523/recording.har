{
  "log": {
    "_recordingName": "buildClusterResourcesRelationships",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "9d73307742273594a28ec2e4efce503e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 148,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "50"
            }
          ],
          "url": "https://192.168.49.2:8443/api/v1/namespaces?limit=50"
        },
        "response": {
          "bodySize": 1867,
          "content": {
            "mimeType": "application/json",
            "size": 1867,
            "text": "{\"kind\":\"NamespaceList\",\"apiVersion\":\"v1\",\"metadata\":{\"selfLink\":\"/api/v1/namespaces\",\"resourceVersion\":\"9161772\"},\"items\":[{\"metadata\":{\"name\":\"default\",\"selfLink\":\"/api/v1/namespaces/default\",\"uid\":\"9c533ce3-b6bd-43e6-947a-f9f73dc297b5\",\"resourceVersion\":\"155\",\"creationTimestamp\":\"2021-05-13T19:34:38Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:38Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-node-lease\",\"selfLink\":\"/api/v1/namespaces/kube-node-lease\",\"uid\":\"11a8778d-4ef0-437b-bb1d-db3cf0a26e82\",\"resourceVersion\":\"33\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-public\",\"selfLink\":\"/api/v1/namespaces/kube-public\",\"uid\":\"ba4e6047-864a-46f9-9f29-3ffe004d7f13\",\"resourceVersion\":\"25\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-system\",\"selfLink\":\"/api/v1/namespaces/kube-system\",\"uid\":\"de2a4eee-505b-49d7-bc08-09026c88dae5\",\"resourceVersion\":\"16\",\"creationTimestamp\":\"2021-05-13T19:34:36Z\",\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2021-05-13T19:34:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:status\":{\"f:phase\":{}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 15:16:26 GMT"
            },
            {
              "name": "content-length",
              "value": "1867"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 146,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T15:16:26.942Z",
        "time": 25,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 25
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
