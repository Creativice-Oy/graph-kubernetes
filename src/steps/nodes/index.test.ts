import { fetchNodes } from '.';
import { createDataCollectionTest } from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { Entities } from '../constants';

describe('#fetchNodes', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'fetchNodes',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchNodes],
      entitySchemaMatchers: [
        {
          _type: Entities.NODE._type,
          matcher: {
            _class: ['Host'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'kube_node' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                name: { type: 'string' },
                displayName: { type: 'string' },
                generation: { type: 'number' },
                deletionGracePeriodSeconds: { type: 'string' },
                resourceVersion: { type: 'string' },
                createdOn: { type: 'number' },
                podCIDR: { type: 'string' },
                podCIDRs: {
                  type: 'array',
                  items: { type: 'string' },
                },
                providerID: { type: 'string' },
                unschedulable: { type: 'boolean' },
                'status.allocatable.cpu': { type: 'string' },
                'status.allocatable.ephemeral-storage': { type: 'string' },
                'status.allocatable.hugepages-1Gi': { type: 'string' },
                'status.allocatable.hugepages-2Mi': { type: 'string' },
                'status.allocatable.memory': { type: 'string' },
                'status.allocatable.pods': { type: 'string' },
                'status.capacity.cpu': { type: 'string' },
                'status.capacity.ephemeral-storage': { type: 'string' },
                'status.capacity.hugepages-1Gi': { type: 'string' },
                'status.capacity.hugepages-2Mi': { type: 'string' },
                'status.capacity.memory': { type: 'string' },
                'status.capacity.pods': { type: 'string' },
                'status.kubeletEndpointPort': { type: 'number' },
                'status.nodeInfo.architecture': { type: 'string' },
                'status.nodeInfo.bootID': { type: 'string' },
                'status.nodeInfo.containerRuntimeVersion': { type: 'string' },
                'status.nodeInfo.kernelVersion': { type: 'string' },
                'status.nodeInfo.kubeProxyVersion': { type: 'string' },
                'status.nodeInfo.kubeletVersion': { type: 'string' },
                'status.nodeInfo.machineID': { type: 'string' },
                'status.nodeInfo.operatingSystem': { type: 'string' },
                'status.nodeInfo.osImage': { type: 'string' },
                'status.nodeInfo.systemUUID': { type: 'string' },
              },
            },
          },
        },
      ],
    });
  });
});
