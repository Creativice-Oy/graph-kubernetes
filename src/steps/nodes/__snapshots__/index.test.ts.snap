// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchNodes should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Host",
      ],
      "_key": "d9e7805e-35e6-4934-bb9a-c0f9853a9ec9",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Node {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": Object {
                "kubeadm.alpha.kubernetes.io/cri-socket": "/var/run/dockershim.sock",
                "node.alpha.kubernetes.io/ttl": "0",
                "volumes.kubernetes.io/controller-managed-attach-detach": "true",
              },
              "clusterName": undefined,
              "creationTimestamp": 2021-04-27T20:57:17.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "beta.kubernetes.io/arch": "amd64",
                "beta.kubernetes.io/os": "linux",
                "kubernetes.io/arch": "amd64",
                "kubernetes.io/hostname": "minikube",
                "kubernetes.io/os": "linux",
                "minikube.k8s.io/commit": "15cede53bdc5fe242228853e737333b09d4336b5",
                "minikube.k8s.io/name": "minikube",
                "minikube.k8s.io/updated_at": "2021_04_27T22_57_20_0700",
                "minikube.k8s.io/version": "v1.19.0",
                "node-role.kubernetes.io/control-plane": "",
                "node-role.kubernetes.io/master": "",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:annotations": Object {
                        "f:kubeadm.alpha.kubernetes.io/cri-socket": Object {},
                      },
                      "f:labels": Object {
                        "f:node-role.kubernetes.io/control-plane": Object {},
                        "f:node-role.kubernetes.io/master": Object {},
                      },
                    },
                  },
                  "manager": "kubeadm",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:20.000Z,
                },
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        "f:minikube.k8s.io/commit": Object {},
                        "f:minikube.k8s.io/name": Object {},
                        "f:minikube.k8s.io/updated_at": Object {},
                        "f:minikube.k8s.io/version": Object {},
                      },
                    },
                  },
                  "manager": "kubectl-label",
                  "operation": "Update",
                  "time": 2021-04-27T20:57:21.000Z,
                },
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:annotations": Object {
                        "f:node.alpha.kubernetes.io/ttl": Object {},
                      },
                    },
                    "f:spec": Object {
                      "f:podCIDR": Object {},
                      "f:podCIDRs": Object {
                        ".": Object {},
                        "v:\\"10.244.0.0/24\\"": Object {},
                      },
                    },
                  },
                  "manager": "kube-controller-manager",
                  "operation": "Update",
                  "time": 2021-05-14T16:22:29.000Z,
                },
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:annotations": Object {
                        ".": Object {},
                        "f:volumes.kubernetes.io/controller-managed-attach-detach": Object {},
                      },
                      "f:labels": Object {
                        ".": Object {},
                        "f:beta.kubernetes.io/arch": Object {},
                        "f:beta.kubernetes.io/os": Object {},
                        "f:kubernetes.io/arch": Object {},
                        "f:kubernetes.io/hostname": Object {},
                        "f:kubernetes.io/os": Object {},
                      },
                    },
                    "f:status": Object {
                      "f:addresses": Object {
                        ".": Object {},
                        "k:{\\"type\\":\\"Hostname\\"}": Object {
                          ".": Object {},
                          "f:address": Object {},
                          "f:type": Object {},
                        },
                        "k:{\\"type\\":\\"InternalIP\\"}": Object {
                          ".": Object {},
                          "f:address": Object {},
                          "f:type": Object {},
                        },
                      },
                      "f:allocatable": Object {
                        ".": Object {},
                        "f:cpu": Object {},
                        "f:ephemeral-storage": Object {},
                        "f:hugepages-1Gi": Object {},
                        "f:hugepages-2Mi": Object {},
                        "f:memory": Object {},
                        "f:pods": Object {},
                      },
                      "f:capacity": Object {
                        ".": Object {},
                        "f:cpu": Object {},
                        "f:ephemeral-storage": Object {},
                        "f:hugepages-1Gi": Object {},
                        "f:hugepages-2Mi": Object {},
                        "f:memory": Object {},
                        "f:pods": Object {},
                      },
                      "f:conditions": Object {
                        ".": Object {},
                        "k:{\\"type\\":\\"DiskPressure\\"}": Object {
                          ".": Object {},
                          "f:lastHeartbeatTime": Object {},
                          "f:lastTransitionTime": Object {},
                          "f:message": Object {},
                          "f:reason": Object {},
                          "f:status": Object {},
                          "f:type": Object {},
                        },
                        "k:{\\"type\\":\\"MemoryPressure\\"}": Object {
                          ".": Object {},
                          "f:lastHeartbeatTime": Object {},
                          "f:lastTransitionTime": Object {},
                          "f:message": Object {},
                          "f:reason": Object {},
                          "f:status": Object {},
                          "f:type": Object {},
                        },
                        "k:{\\"type\\":\\"PIDPressure\\"}": Object {
                          ".": Object {},
                          "f:lastHeartbeatTime": Object {},
                          "f:lastTransitionTime": Object {},
                          "f:message": Object {},
                          "f:reason": Object {},
                          "f:status": Object {},
                          "f:type": Object {},
                        },
                        "k:{\\"type\\":\\"Ready\\"}": Object {
                          ".": Object {},
                          "f:lastHeartbeatTime": Object {},
                          "f:lastTransitionTime": Object {},
                          "f:message": Object {},
                          "f:reason": Object {},
                          "f:status": Object {},
                          "f:type": Object {},
                        },
                      },
                      "f:daemonEndpoints": Object {
                        "f:kubeletEndpoint": Object {
                          "f:Port": Object {},
                        },
                      },
                      "f:images": Object {},
                      "f:nodeInfo": Object {
                        "f:architecture": Object {},
                        "f:bootID": Object {},
                        "f:containerRuntimeVersion": Object {},
                        "f:kernelVersion": Object {},
                        "f:kubeProxyVersion": Object {},
                        "f:kubeletVersion": Object {},
                        "f:machineID": Object {},
                        "f:operatingSystem": Object {},
                        "f:osImage": Object {},
                        "f:systemUUID": Object {},
                      },
                    },
                  },
                  "manager": "kubelet",
                  "operation": "Update",
                  "time": 2021-05-14T16:22:29.000Z,
                },
              ],
              "name": "minikube",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "575600",
              "selfLink": undefined,
              "uid": "d9e7805e-35e6-4934-bb9a-c0f9853a9ec9",
            },
            "spec": V1NodeSpec {
              "configSource": undefined,
              "externalID": undefined,
              "podCIDR": "10.244.0.0/24",
              "podCIDRs": Array [
                "10.244.0.0/24",
              ],
              "providerID": undefined,
              "taints": undefined,
              "unschedulable": undefined,
            },
            "status": V1NodeStatus {
              "addresses": Array [
                V1NodeAddress {
                  "address": "192.168.49.2",
                  "type": "InternalIP",
                },
                V1NodeAddress {
                  "address": "minikube",
                  "type": "Hostname",
                },
              ],
              "allocatable": Object {
                "cpu": "8",
                "ephemeral-storage": "88103912Ki",
                "hugepages-1Gi": "0",
                "hugepages-2Mi": "0",
                "memory": "16289500Ki",
                "pods": "110",
              },
              "capacity": Object {
                "cpu": "8",
                "ephemeral-storage": "88103912Ki",
                "hugepages-1Gi": "0",
                "hugepages-2Mi": "0",
                "memory": "16289500Ki",
                "pods": "110",
              },
              "conditions": Array [
                V1NodeCondition {
                  "lastHeartbeatTime": 2021-06-24T18:34:49.000Z,
                  "lastTransitionTime": 2021-05-14T16:22:29.000Z,
                  "message": "kubelet has sufficient memory available",
                  "reason": "KubeletHasSufficientMemory",
                  "status": "False",
                  "type": "MemoryPressure",
                },
                V1NodeCondition {
                  "lastHeartbeatTime": 2021-06-24T18:34:49.000Z,
                  "lastTransitionTime": 2021-05-14T16:22:29.000Z,
                  "message": "kubelet has no disk pressure",
                  "reason": "KubeletHasNoDiskPressure",
                  "status": "False",
                  "type": "DiskPressure",
                },
                V1NodeCondition {
                  "lastHeartbeatTime": 2021-06-24T18:34:49.000Z,
                  "lastTransitionTime": 2021-05-14T16:22:29.000Z,
                  "message": "kubelet has sufficient PID available",
                  "reason": "KubeletHasSufficientPID",
                  "status": "False",
                  "type": "PIDPressure",
                },
                V1NodeCondition {
                  "lastHeartbeatTime": 2021-06-24T18:34:49.000Z,
                  "lastTransitionTime": 2021-05-14T16:22:29.000Z,
                  "message": "kubelet is posting ready status",
                  "reason": "KubeletReady",
                  "status": "True",
                  "type": "Ready",
                },
              ],
              "config": undefined,
              "daemonEndpoints": V1NodeDaemonEndpoints {
                "kubeletEndpoint": V1DaemonEndpoint {
                  "Port": 10250,
                },
              },
              "images": Array [
                V1ContainerImage {
                  "names": Array [
                    "my-new-image:latest",
                  ],
                  "sizeBytes": 954584458,
                },
                V1ContainerImage {
                  "names": Array [
                    "perl@sha256:0515e9e1c6aecf3a729d4dfb799687c6096c5c1ec117a780952e1555f9c66297",
                    "perl:latest",
                  ],
                  "sizeBytes": 860223906,
                },
                V1ContainerImage {
                  "names": Array [
                    "gcr.io/google_samples/gb-frontend@sha256:50b22839aaf6a18586d6751e8963cf684c27b9873ca926df22cdf88ed4452615",
                    "gcr.io/google_samples/gb-frontend:v3",
                  ],
                  "sizeBytes": 509707703,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/etcd@sha256:4ad90a11b55313b182afc186b9876c8e891531b8db4c9bf1541953021618d0e2",
                    "k8s.gcr.io/etcd:3.4.13-0",
                  ],
                  "sizeBytes": 253392289,
                },
                V1ContainerImage {
                  "names": Array [
                    "kubernetesui/dashboard@sha256:7f80b5ba141bead69c4fee8661464857af300d7d7ed0274cf7beecedc00322e6",
                    "kubernetesui/dashboard:v2.1.0",
                  ],
                  "sizeBytes": 225733746,
                },
                V1ContainerImage {
                  "names": Array [
                    "quay.io/fluentd_elasticsearch/fluentd@sha256:aec118bb3d1c4af358c1d495b14c12781c4ab5e8cfb455edb1ebd2e92750e31d",
                    "quay.io/fluentd_elasticsearch/fluentd:v2.5.2",
                  ],
                  "sizeBytes": 140287084,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/kube-apiserver@sha256:465ba895d578fbc1c6e299e45689381fd01c54400beba9e8f1d7456077411411",
                    "k8s.gcr.io/kube-apiserver:v1.20.2",
                  ],
                  "sizeBytes": 121669114,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/kube-proxy@sha256:326fe8a4508a5db91cf234c4867eff5ba458bc4107c2a7e15c827a74faa19be9",
                    "k8s.gcr.io/kube-proxy:v1.20.2",
                  ],
                  "sizeBytes": 118400203,
                },
                V1ContainerImage {
                  "names": Array [
                    "node@sha256:ed51af876dd7932ce5c1e3b16c2e83a3f58419d824e366de1f7b00f40c848c40",
                    "node:14-alpine",
                  ],
                  "sizeBytes": 116144273,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/kube-controller-manager@sha256:842a071d4ad49b0018f7f7404ac8a4ddfc2bce2ce15b3f8131d89563fda36c9b",
                    "k8s.gcr.io/kube-controller-manager:v1.20.2",
                  ],
                  "sizeBytes": 115852794,
                },
                V1ContainerImage {
                  "names": Array [
                    "node@sha256:9923c9efb13cf7535f67e49b03010f0977a800068e4c8e0e2c93433a6bfa1e77",
                    "node:12-alpine",
                  ],
                  "sizeBytes": 88914929,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/kube-scheduler@sha256:304b3d70497bd62498f19f82f9ef164d38948e5ae94966690abfe9d1858867e2",
                    "k8s.gcr.io/kube-scheduler:v1.20.2",
                  ],
                  "sizeBytes": 46384634,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/coredns@sha256:73ca82b4ce829766d4f1f10947c3a338888f876fbed0540dc849c89ff256e90c",
                    "k8s.gcr.io/coredns:1.7.0",
                  ],
                  "sizeBytes": 45227747,
                },
                V1ContainerImage {
                  "names": Array [
                    "kubernetesui/metrics-scraper@sha256:555981a24f184420f3be0c79d4efb6c948a85cfce84034f85a563f4151a81cbf",
                    "kubernetesui/metrics-scraper:v1.0.4",
                  ],
                  "sizeBytes": 36937728,
                },
                V1ContainerImage {
                  "names": Array [
                    "gcr.io/k8s-minikube/storage-provisioner@sha256:18eb69d1418e854ad5a19e399310e52808a8321e4c441c1dddad8977a0d7a944",
                    "gcr.io/k8s-minikube/storage-provisioner:v5",
                  ],
                  "sizeBytes": 31465472,
                },
                V1ContainerImage {
                  "names": Array [
                    "busybox@sha256:930490f97e5b921535c153e0e7110d251134cc4b72bbb8133c6a5065cc68580d",
                    "busybox:latest",
                  ],
                  "sizeBytes": 1235821,
                },
                V1ContainerImage {
                  "names": Array [
                    "k8s.gcr.io/pause@sha256:927d98197ec1141a368550822d18fa1c60bdae27b78b0c004f705f548c07814f",
                    "k8s.gcr.io/pause:3.2",
                  ],
                  "sizeBytes": 682696,
                },
              ],
              "nodeInfo": V1NodeSystemInfo {
                "architecture": "amd64",
                "bootID": "c6deacb2-5968-4337-926b-31d9acb6450b",
                "containerRuntimeVersion": "docker://20.10.5",
                "kernelVersion": "5.8.0-59-generic",
                "kubeProxyVersion": "v1.20.2",
                "kubeletVersion": "v1.20.2",
                "machineID": "73c9fceff7724090ba72b64bf6e8eff8",
                "operatingSystem": "linux",
                "osImage": "Ubuntu 20.04.1 LTS",
                "systemUUID": "c2652d21-16b4-44d4-8639-f7115c3fb95a",
              },
              "phase": undefined,
              "volumesAttached": undefined,
              "volumesInUse": undefined,
            },
          },
        },
      ],
      "_type": "kube_node",
      "active": undefined,
      "createdOn": 1619557037000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "minikube",
      "generation": undefined,
      "name": "minikube",
      "podCIDR": "10.244.0.0/24",
      "podCIDRs": Array [
        "10.244.0.0/24",
      ],
      "providerID": undefined,
      "resourceVersion": "575600",
      "status.allocatable.cpu": "8",
      "status.allocatable.ephemeral-storage": "88103912Ki",
      "status.allocatable.hugepages-1Gi": "0",
      "status.allocatable.hugepages-2Mi": "0",
      "status.allocatable.memory": "16289500Ki",
      "status.allocatable.pods": "110",
      "status.capacity.cpu": "8",
      "status.capacity.ephemeral-storage": "88103912Ki",
      "status.capacity.hugepages-1Gi": "0",
      "status.capacity.hugepages-2Mi": "0",
      "status.capacity.memory": "16289500Ki",
      "status.capacity.pods": "110",
      "status.kubeletEndpointPort": 10250,
      "status.nodeInfo.architecture": "amd64",
      "status.nodeInfo.bootID": "c6deacb2-5968-4337-926b-31d9acb6450b",
      "status.nodeInfo.containerRuntimeVersion": "docker://20.10.5",
      "status.nodeInfo.kernelVersion": "5.8.0-59-generic",
      "status.nodeInfo.kubeProxyVersion": "v1.20.2",
      "status.nodeInfo.kubeletVersion": "v1.20.2",
      "status.nodeInfo.machineID": "73c9fceff7724090ba72b64bf6e8eff8",
      "status.nodeInfo.operatingSystem": "linux",
      "status.nodeInfo.osImage": "Ubuntu 20.04.1 LTS",
      "status.nodeInfo.systemUUID": "c2652d21-16b4-44d4-8639-f7115c3fb95a",
      "unschedulable": undefined,
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "kube_node",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
